library/core/src/lib.rs

```diff
@@ -24,9 +24,8 @@
 //!   which are generated by Rust codegen backends. Additionally, this library can make explicit
 //!   calls to `strlen`. Their signatures are the same as found in C, but there are extra
 //!   assumptions about their semantics: For `memcpy`, `memmove`, `memset`, `memcmp`, and `bcmp`, if
-//!   the `n` parameter is 0, the function is assumed to not be UB. Furthermore, for `memcpy`, if
-//!   source and target pointer are equal, the function is assumed to not be UB.
-//!   (Note that these are standard assumptions among compilers:
+//!   the `n` parameter is 0, the function is assumed to not be UB, even if the pointers are NULL or
+//!   dangling. (Note that making extra assumptions about these functions is common among compilers:
```

Comment:
While removing the "equal" part, you added "NULL or dangling" -- isn't that a *new* assumption? Or are we already living by that elsewhere?

Reply:
That's a clarification. We already have "if the n parameter is 0, the function is assumed to not be UB", which was already meant to include *all* pointers, including NULL or dangling. I just made it more explicit.